### PR TITLE
run new shoot clusters without overlay if not explicitly specified

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/area networking
/kind enhancement
/platform alicloud

**What this PR does / why we need it**:
A mutating webhook is added, so that all new calico shoot clusters on alicloud will be created without an overlay if not explicitly specified in the shoot spec.
Prerequisite for https://github.com/gardener/backlog/issues/29.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
All new calico alicloud shoot clusters will be created without an overlay if not explicitly specified in the shoot spec.
```

### CODE REVIEW

**Comment by ScheererJ** [comment]
Rebased as requested.

File: pkg/webhook/network/mutate.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+	calicov1alpha1 "github.com/gardener/gardener-extension-networking-calico/pkg/apis/calico/v1alpha1"
+	calicov1alpha1helper "github.com/gardener/gardener-extension-networking-calico/pkg/apis/calico/v1alpha1/helper"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+)
+
+func mutateNetworkConfig(new, old *extensionsv1alpha1.Network) error {
```
Comment (shaoyongfeng) [inline pkg/webhook/network/mutate.go:28]: what will happen if shoot uses cilium network instead of calico? will networkconfig also be mutated?

**Review by shaoyongfeng — APPROVED** [review]

**Pull request merged**